### PR TITLE
[node.js] express4 app linter [APPSEC-10464]

### PR DESCRIPTION
**Pull request by iunanua**
## Description

Adds eslint config file, dependencies and fixes linter errors

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: utils/build/docker/nodejs/express4/iast/index.js
Comment (iunanua) [inline utils/build/docker/nodejs/express4/iast/index.js:84]: Why is there a " in the tag value `'appsec.event"'`? :S

File: utils/build/docker/nodejs/express4/.eslintrc.json
Comment (simon-id) [inline utils/build/docker/nodejs/express4/.eslintrc.json:1]: did you get this from dd-trace ?

File: utils/build/docker/nodejs/express4/app.js
Comment (simon-id) [inline utils/build/docker/nodejs/express4/app.js:1]: disable it directly in the eslint config file i'd say

File: utils/build/docker/nodejs/express4/app.js
Comment (simon-id) [inline utils/build/docker/nodejs/express4/app.js:190]: would that work ?
```suggestion
  require('glob')
```

File: utils/build/docker/nodejs/express4/.eslintrc.json
Comment (iunanua) [reply utils/build/docker/nodejs/express4/.eslintrc.json:1]: [from the system-tests nextjs app](https://github.com/DataDog/system-tests/blob/main/utils/build/docker/nodejs/nextjs/.eslintrc.json)

Comment (iunanua) [reply utils/build/docker/nodejs/express4/.eslintrc.json:1]: but probably it comes from dd-trace

File: utils/build/docker/nodejs/express4/app.js
Comment (simon-id) [inline utils/build/docker/nodejs/express4/app.js:244]:
```suggestion
  const serviceName = req.query.serviceName
```

File: utils/build/docker/nodejs/express4/iast/index.js
Comment (simon-id) [reply utils/build/docker/nodejs/express4/iast/index.js:84]: `appsec.event: true` is used to signal a WAF attack has been detected. Now why is it used here i have no idea haha

Comment (iunanua) [reply utils/build/docker/nodejs/express4/iast/index.js:84]: it will have no effect anyway because the tests are setting `'appsec.event"'` and not `'appsec.event'`

File: utils/build/docker/nodejs/express4/integrations/db/postgres.js
Comment (simon-id) [inline utils/build/docker/nodejs/express4/integrations/db/postgres.js:20]: Heads up: that's not a linting change

File: utils/build/docker/nodejs/express4/integrations/db/postgres.js
Comment (simon-id) [inline utils/build/docker/nodejs/express4/integrations/db/postgres.js:33]: potential side effect here

File: utils/build/docker/nodejs/express4/integrations/db/postgres.js
Comment (iunanua) [reply utils/build/docker/nodejs/express4/integrations/db/postgres.js:20]: it's the `prefer-promise-reject-errors` rule.
Do you mean it could change the app behavior?

File: utils/build/docker/nodejs/express4/integrations/db/postgres.js
Comment (iunanua) [reply utils/build/docker/nodejs/express4/integrations/db/postgres.js:33]: I don't think so. 
The rest of the methods do not create an instance and tests are passing without it.

File: utils/build/docker/nodejs/express4/iast/index.js
Comment (simon-id) [reply utils/build/docker/nodejs/express4/iast/index.js:84]: Oh shit I didn't even see the double quote here, that's probably not right. Which also probably means that this setTag is useless

Comment (simon-id) [reply utils/build/docker/nodejs/express4/iast/index.js:84]: @uurien the git blame points at you

File: utils/build/docker/nodejs/express4/integrations/db/postgres.js
Comment (simon-id) [reply utils/build/docker/nodejs/express4/integrations/db/postgres.js:33]: true, i'll let the APM team decide

File: utils/build/docker/nodejs/express4/iast/index.js
Comment (uurien) [reply utils/build/docker/nodejs/express4/iast/index.js:84]: wow, it seems that it is wrong, and we should remove these lines. Probably copied from java test (I checked that they are adding `appsec.event=true`, but it doesn't make any sense.
And as mentioned before... it has a typo with the last `"` ...

**Review by simon-id — APPROVED** [review]
LGTM but we should still wait for APM approval too

File: utils/build/docker/nodejs/express4/integrations/db/postgres.js
Comment (simon-id) [reply utils/build/docker/nodejs/express4/integrations/db/postgres.js:20]: Yes, idk if it's important or not, but just a heads up for APM to check if they actually want this change or not

**Comment by uurien** [comment]
You have added the file, that is great. But I think that we should add a task in the image building or CI process to check that the new code is not breaking the linter.

**Comment by robertomonteromiguel** [comment]
> You have added the file, that is great. But I think that we should add a task in the image building or CI process to check that the new code is not breaking the linter.

+1

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
As Ugaitz says: You have added the file, that is great. But I think that we should add a task in the image building or CI process to check that the new code is not breaking the linter.

File: utils/build/docker/nodejs/express4/iast/index.js
Comment (iunanua) [reply utils/build/docker/nodejs/express4/iast/index.js]: removed

**Comment by iunanua** [comment]
done, but I have added two new steps in the `lint_code` action:
- one to set up node in the runner. Is it too much invasive? If there are problems with that I could skip the installation of node and just check if npm exists before running npm...
- and another one to run npm lint

**Review by robertomonteromiguel — APPROVED** [review]

**Comment by simon-id** [comment]
Not sure it's an amazing idea to put the lint job in this file. I'll let charles decide where's the best place

**Comment by iunanua** [comment]
@cbeauchesne wdyt?

**Comment by cbeauchesne** [comment]
> Not sure it's an amazing idea to put the lint job in this file. I'll let charles decide where's the best place

@simon-id what is the alternative ?

File: .github/actions/lint_code/action.yaml
Comment (cbeauchesne) [inline .github/actions/lint_code/action.yaml:34]: What about installing only lint deps to  speed up the job ?

**Comment by simon-id** [comment]
@cbeauchesne no idea, just looks weird to add something specific to node in here

**Comment by cbeauchesne** [comment]
No, it's ok for me, I prefer to have in in the lint job : 

* we do not spawn more than 130 worker for nothing
* user are warned sooned

File: .github/actions/lint_code/action.yaml
Comment (iunanua) [reply .github/actions/lint_code/action.yaml]: done!

Comment (simon-id) [reply .github/actions/lint_code/action.yaml]: I see you added `-D`. Did you test it to make sure it was only installing the dev deps ? I just tried it and I don't think it's doing that ?

Comment (iunanua) [reply .github/actions/lint_code/action.yaml]: There seem to have been different ways: `--dev`, `--only=dev`, `--production=false` and `-D` and I've read comments stating that `-D` was working with the latest npm versions

But I haven't checked it :S

Comment (iunanua) [reply .github/actions/lint_code/action.yaml]: There may not be a way to install only the dev dependencies...

**Review by uurien — APPROVED** [review]

File: .github/actions/lint_code/action.yaml
Comment (simon-id) [reply .github/actions/lint_code/action.yaml]: Back to doing `npm install eslint eslint-standard....` then i guess

Comment (simon-id) [reply .github/actions/lint_code/action.yaml]: what about using the global arg ? `-g` that shouldn't use any package/lock file no ?

Comment (uurien) [reply .github/actions/lint_code/action.yaml]: I don't know if it could work with the other dependencies, `eslint` yes, but `eslint-plugin-mocha`,... I think that they should be in `node_modules`.
I mentioned to Igor to try with something like:
```shell
cp package.json package-back.json && node -pe "const pack = require('./package.json');delete pack.dependencies;require('fs').writeFileSync('./package.json',JSON.stringify(pack))" && npm install --no-package-lock axios && mv package-back.json package.json
```

It should install only `devDependencies`, because we are removing `dependencies` from `package.json`.
Just to see how much time can we earn, if the time difference is insignificant... we can just maintain the normal `npm install`

Comment (iunanua) [reply .github/actions/lint_code/action.yaml]: Around the same time:
- npm install -> 41s
- your hack -> 43s

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**
